libraries/AP_L1_Control/AP_L1_Control.cpp

```diff
@@ -145,36 +136,41 @@ float AP_L1_Control::turn_distance(float wp_radius, float turn_angle) const
     return distance_90 * turn_angle / 90.0f;
 }
 
-float AP_L1_Control::loiter_radius(const float radius) const
+float AP_L1_Control::_calc_min_turn_radius(float indicated_airspeed,
```

Comment:
Rather than the NaN, have two finction with a difrenct fingerprint. One takes two floats is the main funciton. Then have one with the same name that takes one float then fetched the alt and passes it into the other function.